Refactor getComponentName function

kenzhemir opened the pull request:
## Why?
I noticed this code:
```
try {
  throw new Error('...')
} catch (error) {
  ...
}
```

I think there is no need to throw Error object just to get error stack, it's enough to create Error object and use it, like so:
```
const error = new Error('...')
```

Additionally, [Error.prototype.stack is non-standard](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/Stack) and there might be a situation where error.stack is `undefined`, so I added test for this case (previously it was ignored from coverage). In order to do that, I had to move error as optional first argument to this function. There might be other ways to do it, but this was the easiest one (couldn't correctly override global Error constructor)

kenzhemir committed:
Refactor getComponentName function

There is no need to throw Error object just to get error stack, it's
enough to just create Error object and use it.
Additionally, Error.prototype.stack is non-standard so there might be a
situation where stack is undefined, so I added test for this case.

dolfbarr commented:
Great catch & great work! Thank you for your contribution!

_nit, non-blocker:_ it would be also nice to add a small test to cover a stack parsing (line 81) as well since coverage ignore is removed; it was not added initially just because it's a little bit 'hacky' way to gather component name via non-standard `Error` feature.

kenzhemir commented:
I can take a look tomorrow into parsing part👍

kenzhemir committed:
Add test when stack format is not supported

- Added test where error.stack has some format that getComponentName
doesn't expect
- Changed the return value from "undefined" to "" in such situations

dolfbarr commented:
LGTM! Great work!
